sycl/source/detail/graph_impl.hpp

```diff
@@ -1198,6 +1200,9 @@ class graph_impl : public std::enable_shared_from_this<graph_impl> {
   std::map<std::weak_ptr<sycl::detail::queue_impl>, std::shared_ptr<node_impl>,
            std::owner_less<std::weak_ptr<sycl::detail::queue_impl>>>
       MBarrierDependencyMap;
+
+  unsigned long long MID;
+  inline static std::atomic<unsigned long long> NextAvailableID = 0;
```

Comment:
Nit but I think it's worth having a comment for these that they are only used for `std::hash`